### ISSUE TITLE
Naming views folders

### ISSUE THREAD

**Issue by AlexNodex** [mozilla/nunjucks#831]
Hi.

First off - great project, I am an avid twig user and the two are very similar.

In twig you can name your "views" paths so that one can get to the correct file with a prefix.

In twig it's something similar to `AddLoader('path/to/views','foo');` then in my app I can prefix an include file like so 
`{% include '@foo/index.html' %}`

Twig then knows that the file I am asking for is `path/to/views/index.html` (it basically translates @foo to the filesystem path)

I notice this is missing or undocumented in nunjucks?

It would be a strong addition to the project as it allows namespacing of view folders and one to be able to resolve the view without knowledge of the filesystem only needing the @prefix

**Comment by jbmoelker** [comment]
@AlexNodex are "named views" part of Twig? I don't see it mentioned under [Twig_Loader_FileSystem](http://twig.sensiolabs.org/doc/api.html#twig-loader-filesystem). I do see it described in [Namespaced Twig Paths in Symfony](http://symfony.com/doc/current/templating/namespaced_paths.html). As such I don't think it should be part of Nunjucks, but rather it could be an extension of Nunjucks, similar to how Symfony extends Twig.

If you want to achieve something similar in Nunjucks today, you can use a [global variable](https://mozilla.github.io/nunjucks/api.html#addglobal) in combination with string concatenation:

```javascript
env.addGlobal('@foo', 'some/view/alias/')
```
And in the template:
```html
{% include @foo + 'index.html' %}
```

Here's a [working demo](http://plnkr.co/edit/rlqOD7?p=preview).

**Comment by AlexNodex** [comment]
they must be part of twig as I use twig and not symphony and I'm able to use them. I will dig out the docs on them for you

**Comment by AlexNodex** [comment]
http://twig.sensiolabs.org/doc/api.html#built-in-loaders

If you look at the addPath method you can see the namespacing they employ which is 'path/to/views','namespace'

**Comment by jbmoelker** [comment]
@AlexNodex you're totally right. I didn't know about that Twig feature. I'm curious if it's also in Jinja2.

The [documentation you refer to](http://twig.sensiolabs.org/doc/api.html#built-in-loaders) uses an associative array:
```php
$twig = new Twig_Environment($loader, array(
    'cache' => '/path/to/compilation_cache',
));
```
In JavaScript that would translate to an object. So I would imagine the Nunjucks way could be:
```javascript
env = new nunjucks.Environment(new nunjucks.FileSystemLoader({
  cache: '/path/to/compilation_cache'
}));
```
Note: would be the same for the WebLoader.

I can imagine this being a useful feature. I don't know anyone willing to pick this up, but I would welcome a PR :)

**Comment by AlexNodex** [comment]
The main trouble is that it would be a breaking change, certainly the addPath part as it currently only takes a single dimensional array and not an object

**Comment by jbmoelker** [comment]
I agree we shouldn't introduce breaking changes for this. We should be able to test if input is string, array ([`Array.isArray(input)`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray)) or Object. In which case it would not break any of the existing functionality (no breaking change) and would simply be an additional feature.

**Comment by AlexNodex** [comment]
I'll see if I can knock up some code and see if it can work
